### ISSUE TITLE
PathingThread wasted

### ISSUE THREAD

**Issue by jzy-chitong56** [SMUnlimited/AMAI#132]
In general maps, neutral buildings are invincible and unmanned, they cannot move , so no need for secondary location detection

when PathingThread detection completed ， The thread is wasted , Re detection requires excrete location and groups , and when secondary location detection will create Pathing unit again , causing the game stuck 

so maybe can give a loop 

```
loop
if nearest_neutral[GetHealthFountainID()] != null then
  call NEUTRAL_ENEMY_CHECK(NEUTRAL_HEALING_FOUNTAIN)
endif
if nearest_neutral[GetManaFountainID()] != null then
  call NEUTRAL_ENEMY_CHECK(NEUTRAL_MANA_FOUNTAIN)
endif
if neutral_guarded[i] then
  call NEUTRAL_GUARDED(i)
endif
call waterminecheck()
set i = i + 1
sleep(80)
endloop
```

then the check job no longer run on JOB Thread , Save JOB's operations

**call TQAddJob(10, NEUTRAL_ENEMY_CHECK, NEUTRAL_HEALING_FOUNTAIN)
call TQAddJob(10, NEUTRAL_ENEMY_CHECK, NEUTRAL_MANA_FOUNTAIN)
call TQAddJob(30 * sleep_multiplier, NEUTRAL_GUARDED, nn)** 

yes, PathingThread need longer time check all build , Run NEUTRAL_GUARDED and NEUTRAL_ENEMY_CHECK  when check done,  really unreasonable , but anyway , no need for secondary location detection

```
loop
call waterminecheck()
sleep(80)
endloop
```

--------------------------------------------------------------------------------

Another issue is that water mine or water neutral buildings
some build no water , they just blocked by doors or trees , maybe human player can road has been cleared , so need secondary location detection , can give a function waterminecheck() , but I haven't thought out the specific code yet
